### PR TITLE
Enable usage in no_std contexts

### PR DESCRIPTION
**Pull request by cmr**
Please also publish the new version to crates.io :)

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [package]
 name = "byteorder"
-version = "0.5.1"  #:version
+version = "0.5.2"  #:version
```
Comment (BurntSushi) [inline Cargo.toml]: Could you remove the version bump please? I like to do that in a separate commit/tag.

**Comment by BurntSushi** [comment]
@cmr This broadly looks good. Why did you change the feature from `no-std` to `std`? I think that implies a breaking change, which would require a `0.6` release to get this change released. If we stick with the `no-std` feature (which seems simpler because you don't have to muck with default features?), then I think this change has no breaking changes, yes?

**Comment by cmr** [comment]
The feature wasn't actually used at all. And the positive form is more
suitable because cargo features are formed from the union of all
features needed by dependencies; if one dep wants to use a std dependent
feature implicitly, but another requires the no std feature, the crate
will be compiled no std, breaking the first crate.
 
 
On Mon, May 2, 2016, at 01:49, Andrew Gallant wrote:
> @cmr This broadly looks good. Why did you change the feature from `no-
> std` to `std`? I think that implies a breaking change, which would
> require a `0.6` release to get this change released. If we stick with
> the `no-std` feature (which seems simpler because you don't have to
> muck with default features?), then I think this change has no breaking
> changes, yes?
>
> ---
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub:
> https://github.com/BurntSushi/byteorder/pull/46#issuecomment-216051310
 
 
--
cmr
+610481782084
http://octayn.net/

**Comment by BurntSushi** [comment]
Ah good point!

The feature `no-std` appears to have been used though? It was guarding the definition of the `new` module.

**Comment by cmr** [comment]
Oops, right, it just didn't enable `no_std`.

**Comment by BurntSushi** [comment]
Looks great! I'll get a `0.6` release out soon.
